gandalf.js

```diff
@@ -1,129 +1,125 @@
 /**
```

Comment:
SCRIPT-This function might benefit from using async/await instead of callbacks for better readability.